src/ast/query.rs

```diff
@@ -299,11 +299,19 @@ impl fmt::Display for Select {
             write!(f, " WHERE {selection}")?;
         }
         match &self.group_by {
-            GroupByExpr::All => write!(f, " GROUP BY ALL")?,
-            GroupByExpr::Expressions(exprs) => {
+            GroupByExpr::All(modifiers) => {
+                write!(f, " GROUP BY ALL")?;
+                if !modifiers.is_empty() {
+                    write!(f, " {}", display_separated(modifiers, " "))?;
+                }
+            }
+            GroupByExpr::Expressions(exprs, modifiers) => {
                 if !exprs.is_empty() {
                     write!(f, " GROUP BY {}", display_comma_separated(exprs))?;
                 }
+                if !modifiers.is_empty() {
+                    write!(f, " {}", display_separated(modifiers, " "))?;
+                }
             }
```

Comment:
this seems like its identical to the display method on GroupByExpr, if so would it make sense to simplify a bit by calling display instead?